application/models/QuestionBaseRenderer.php

```diff
@@ -61,8 +61,23 @@ public function __construct($aFieldArray, $bRenderDirect = false)
 
         $questionScript = $this->getQuestionScript();
         if (!empty($questionScript)) {
-            $sScriptRendered = LimeExpressionManager::ProcessString($questionScript, $this->oQuestion->qid, ['QID' => $this->oQuestion->qid]);
+            /* Replace Expression Manager */
+            $sScriptRendered = LimeExpressionManager::ProcessString(
+                $questionScript,
+                $this->oQuestion->qid,
+                [
+                    'QID' => $this->oQuestion->qid,
+                    'GID' => $this->oQuestion->gid,
+                    'SGQ' => $this->oQuestion->sid . "X" . $this->oQuestion->gid . "X" . $this->oQuestion->qid,
+                ]
+            );
             $this->addScript('QuestionStoredScript-' . $this->oQuestion->qid, $sScriptRendered, LSYii_ClientScript::POS_POSTSCRIPT);
+            /* Reset needed replacement */
+            LimeExpressionManager::updateReplacementFields(array(
```

Comment:
Sorry, this comment was left on "pending".

Reply:
See for Question : https://github.com/LimeSurvey/LimeSurvey/blob/ab2d67a374b57173505cf77570fb17c57653d494/application/helpers/SurveyRuntimeHelper.php#L496

and Group : https://github.com/LimeSurvey/LimeSurvey/blob/ab2d67a374b57173505cf77570fb17c57653d494/application/helpers/SurveyRuntimeHelper.php#L520

https://github.com/LimeSurvey/LimeSurvey/commit/9c055e54b013833fb905b023787ae93291e5130f

Reply:
I know we have been using it.
I was just thinking maybe there is a risk in adding it now. Some unexpected behaviour we are not testing right now.

Still, having mentioned that, if you feel OK, let's move forward :)

Reply:
>  I was just thinking maybe there is a risk in adding it now. Some unexpected behaviour we are not testing right now.

You mean adding QID,SGQ and GID in script ? 

Or reset QID ? (since SGQ and GID was not set …) if you want i can remove tghe QID reste, but i really think it must be resetted until we have a way to ,check we are in Questin by question mode (for example)

Currently : if you look at issue : SGQ and GID is set when you preview survey as admin … but not for public user … have sale for public and allways is always better …

Reply:
> You mean adding QID,SGQ and GID in script ?

I mean adding this now
https://github.com/LimeSurvey/LimeSurvey/pull/2855/files#diff-39aef150f7fc0d3ff6e986ecb9a1c8ac95a71c109a87eb378e4a3607d8f665d2R76

> if you want i can remove tghe QID reste, but i really think it must be resetted until we have a way to ,check we are in Questin by question mode (for example)

It is up to you. I wouldn't add it on this commit. But on another one as to have it properly assessed and tested.
Right now, the focus on the testing seems on reviewing the placeholders replacement, not on the reset statement.

Reply:
We reset it for question (Navigator didn't have it).

I really think it's better here.

Reply:
Go ahead. It is already approved. Thanks!